pandas/io/json/_json.py

```diff
@@ -655,11 +666,16 @@ def __init__(
         self.chunksize = chunksize
         self.nrows_seen = 0
         self.should_close = False
+        self.nrows = nrows
 
         if self.chunksize is not None:
             self.chunksize = _validate_integer("chunksize", self.chunksize, 1)
             if not self.lines:
                 raise ValueError("chunksize can only be passed if lines=True")
+        if self.nrows is not None:
+            self.nrows = _validate_integer("nrows", self.nrows, 0)
+            if not self.lines:
+                raise ValueError("nrows can only be passed if lines=True")
```

Comment:
Thanks @hasnain2808 

Could you add a test which hits this line, i.e. that checks that this ValueError is raised if `nrows` is set but `lines` isn't?

Reply:
Thanks @MarcoGorelli for the review.
I added the test. Tested the test by commenting out the appropriate lines and pushed the commit

Reply:
Great, thanks - can you also add the github issue number as a comment at the top of each test you've added? Something like
```python
def test_readjson_nrows(nrows):
    # GH<issue number goes here>
```
?

Reply:
Added @MarcoGorelli